package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -0,0 +1,474 @@
+        local_origins: array, shape (N-2, 3)
+            the projected origin for each helix
+        all_bends: array, shape (N-3, N-3)
+            angles between each local axis
+        global_axis: array, shape (3,)
+            vector of best fit through origins, pointing at the first origin.
+        local_screw_angles: array, shape (N-2,)
+            cylindrical azimuth angle to plane of global_axis and ref_axis
+    """
+
+    #          ^               ^
+    #           \             / bi
+    #            \           /
+    #         CA_i+2 <----- CA_i+1
+    #         /    \       /   ^
+    #        /    r \     /     \
+    #     V /        \ θ /       \
+    #      /          \ /       CA_i
+    #     v           origin
+    #   CA_i+3
+    #
+    # V: vectors
+    # bi: bisectors. Note: not real bisectors are the vectors aren't normalised
+    # θ: local_twists
+    # origin: origins
+    # local_axes: in the plane of the screen. Orthogonal to the "bisectors"
+
+    vectors = positions[1:] - positions[:-1]  # (n_res-1, 3)
+    bisectors = vectors[:-1] - vectors[1:]  # (n_res-2, 3)
+    bimags = mdamath.pnorm(bisectors)  # (n_res-2,)
+    adjacent_mag = bimags[:-1] * bimags[1:]  # (n_res-3,)
+
+    # find angle between bisectors for twist and n_residue/turn
+    cos_theta = mdamath.pdot(bisectors[:-1], bisectors[1:])/adjacent_mag
+    cos_theta = np.clip(cos_theta, -1, 1)
+    twists = np.arccos(cos_theta)  # (n_res-3,)
+    local_twists = np.rad2deg(twists)
+    local_nres_per_turn = 2*np.pi / twists
+
+    # find normal to bisectors for local axes
+    cross_bi = np.cross(bisectors[:-1], bisectors[1:])  # (n_res-3, 3)
+    local_axes = (cross_bi.T / mdamath.pnorm(cross_bi)).T  # (n_res-3, 3)
+    local_axes = np.nan_to_num(local_axes)
+
+    zero_vectors = np.tile(np.any(local_axes, axis=1), (len(local_axes), 1)).T
+    # find angles between axes for bends
+    bend_theta = np.matmul(local_axes, local_axes.T)  # (n_res-3, n_res-3)
+    # set angles to 0 between zero-vectors
+    bend_theta = np.where(zero_vectors+zero_vectors.T,  # (n_res-3, n_res-3)
+                          bend_theta, 1)
+    bend_matrix = np.rad2deg(np.arccos(np.clip(bend_theta, -1, 1)))
+    # local bends are between axes 3 windows apart
+    local_bends = np.diagonal(bend_matrix, offset=3)  # (n_res-6,)
+
+    # radius of local cylinder
+    radii = (adjacent_mag**0.5) / (2*(1.0-cos_theta))  # (n_res-3,)
+    # special case: angle b/w bisectors is 0 (should virtually never happen)
+    # guesstimate radius = half bisector magnitude
+    radii = np.where(cos_theta != 1, radii, (adjacent_mag**0.5)/2)
+    # height of local cylinder
+    heights = np.abs(mdamath.pdot(vectors[1:-1], local_axes))  # (n_res-3,)
+
+    local_helix_directions = (bisectors.T/bimags).T  # (n_res-2, 3)
+
+    # get origins by subtracting radius from atom i+1
+    origins = positions[1:-1][:]  # (n_res-2, 3)
+    origins[:-1] -= (radii*local_helix_directions[:-1].T).T
```

Comment:
I think this needs to use `copy()` rather than `[:]` to stop the input `positions` array from being altered here